### ISSUE TITLE
Ampache api always returning empty contents with amarok

### ISSUE THREAD

**Issue by aleixq** [owncloud/music#854]
Amarok player always adds the "limit=0" when fetching the collection via ampache. And when limit=0 is there ampache endpoint returns 
```
<?xml version="1.0" encoding="UTF-8"?>
<root>
  <total_count>0</total_count>
</root>
```
Testing the same with curl happens the same.

```
curl "https://example.com/apps/music/ampache/server/json.server.php?auth=xxxxxxxxxxx&action=artists&limit=1&filter=django"
[]
```

Not checking the type when testing if 0 in https://github.com/owncloud/music/blob/76399da5f8b2cc9c0048d251b8a3162e2ef64251/lib/Controller/AmpacheController.php#L883  solves the issue:
```

        private static function validateLimitOrOffset($value) : ?int { 
                //print_r(strval($value) == 0? "zero":"nozero"); 
-                if (\ctype_digit(\strval($value)) && $value !== 0) { 
+               if (\ctype_digit(\strval($value)) && $value != 0) {
                        return \intval($value);
                } else {
                        return null;
                }
        }
```

**Comment by paulijar** [comment]
Okay, thanks for digging. I feel that Amarok is at fault here, as the Ampache API documentation does not state that the value `limit=0` would have any special purpose, instead of being interpreted literally. What it does say, is that `limit=none` could be used to unset the limit. But as the real Ampache server seems to disregard also the 0 limit arguments, I guess we need to follow.

Btw, are you using the Linux version of Amarok? Because I have never managed to make the Windows version of the app connect my cloud. It seems to communicate with a malformed URL. I suspect that this is because the Win version has not been updated for ages (it's still 2.8.0 which was apparently released in 2013).

**Comment by aleixq** [comment]
That's right Paul, sorry for not reading the note about limit in ampache api docs... So I guess that this bug must be enqueued in amarok bugtracker... 

About the amarok version I use: Yes it's the last linux version (2.9.71) , and It's working fine with nc music ampache api. 
I came to the good and old amarok because  I cannot find any desktop player that connects to ampache in gnu/linux... But is just to inspect a good ampache client implementation to look for ideas to create a new ampache/subsonic player using kde's kirigami.

Thank's Paul It can be closed .

**Comment by paulijar** [comment]
The fix for this problem has now been released in Music v1.2.0.
